kiali/kiali
# Autodetect istiod deployment name

### Describe the change

Autodetect istiod deployment name instead of relying on the config value.

Also removes the individual istiod pods from the istio status. If istiod pods are unhealthy it will still reflect in the status since this comes from the deployment status. If Kiali can't port forward to the istiod pods then it will update the status as unreachable. Moves the check for kiali connecting to the istiod pods to the `istio.Discovery.Mesh` method.

### Steps to test the PR

1. Deploy kiali and istio
2. Set `external_services.istio.deployment_name = ""`
3. Ensure kiali still works

Test Multiple revisions with sail:

1. Deploy istio with sail: https://github.com/istio-ecosystem/sail-operator?tab=readme-ov-file#deploying-the-operator
2. Deploy `Istio` resource: 
    ```
    # Namespace must exist prior to creating istio resource
    kubectl get ns istio-system || kubectl create ns istio-system
    kubectl apply -f - <<EOF
    apiVersion: operator.istio.io/v1alpha1
    kind: Istio
    metadata:
      name: default
    spec:
      version: v1.22.0
      namespace: istio-system
      updateStrategy:
        type: RevisionBased
        inactiveRevisionDeletionGracePeriodSeconds: 3000
    EOF
    ```
3. Deploy kiali operator and kiali
    ```
    make build build-ui cluster-push operator-create kiali-create
    ```
4. Ensure kiali works
5. Change the version of the `Istio` resource
6. Navigate to the mesh page in kiali and verify you see both istiods, one for each revision

### Automation testing

Unit testing included

### Issue reference

Relates to #7177 

Operator PR: https://github.com/kiali/kiali-operator/pull/784
